fiftyone/core/collections.py

```diff
@@ -945,6 +945,16 @@ def _get_default_sample_fields(
                 use_db_fields=use_db_fields,
             )
 
+            if self.media_type == fom.POINT_CLOUD and issubclass(
```

Comment:
We'll need to generalize the `self.media_type == fom.POINT_CLOUD` check a bit to handle the different cases that can arise involving grouped datasets. Basically we need the same logic as [`_contains_videos()`](https://github.com/voxel51/fiftyone/blob/60d3c2c0334777a7bc73e2e0641902ac2dbdb227/fiftyone/core/collections.py#L9484) for `POINT_CLOUD` media type rather than `VIDEO`.

I'd recommend adding a new `_contains_media_type()` method like so:

```py
def _contains_media_type(self, media_type, any_slice=False):
   # generalize the existing _contains_videos() here

def _contains_videos(self, any_slice=False):
    return self._contains_media_type(fom.VIDEO, any_slice=any_slice)
```

For the purposes of this PR, the check we need is then `self._contains_media_type(fom.POINT_CLOUD, any_slice=True)`.

Also, for generality, the logic we're discussing here should be added to `_get_default_frame_fields()` as well, for when users can have playback of 3D scenes 😄

Reply:
Side note: I considered suggesting that we move this logic inside `EmbeddedDocumentField._get_default_fields()`. This would cause the rest of the SDK to know that these fields are required, which would prevent users from doing certain things like:

```py
# Delete the field
dataset.delete_sample_field("ground_truth.detections.location")

# Remove the field from the dataset's schema without clearing the values
dataset.remove_dynamic_sample_field("ground_truth.detections.location")
```

on a dataset that contains 3D data.

For context, since these 3D-specific fields are not formally declared on the dataset's schema by default, if users want them to be available in the App then they have to manually declare them like so:

```py
dataset.add_sample_field("ground_truth.detections.location", fo.ListField, subfield=fo.FloatField)
dataset.add_sample_field("ground_truth.detections.dimensions", fo.ListField, subfield=fo.FloatField)
dataset.add_sample_field("ground_truth.detections.rotation", fo.ListField, subfield=fo.FloatField)
```

For reversibility, it's probably best to let them be able to undeclare/delete these fields from their schema if they want to.

**tldr**: I think it is best to *NOT* move the logic from this PR inside `EmbeddedDocumentField._get_default_fields()`.